src/data/bucket/line_bucket.js

```diff
@@ -119,8 +120,11 @@ class LineBucket implements Bucket {
         this.layers = options.layers;
         this.layerIds = this.layers.map(layer => layer.id);
         this.index = options.index;
-        this.features = [];
         this.hasPattern = false;
+        this.patternFeatures = [];
+
+        const sortKey = this.layers[0].layout.get('line-sort-key');
+        this.sortFeaturesByKey = sortKey.constantOr(1) !== undefined;
```

Comment:
I see this pattern of `constantOr` used elsewhere, but don't understand how/when it is expected to evaluate to `false` if `constantOr` returns `1` as the fallback value

More importantly, what is the need for this instance variable? In `populate()`, cant `lineSortKey` be used to determine if the features need to be sorted?

Reply:
Removed the instance variable in CircleBucket and will modify this patchset soon to remove from Fill and Line. `constantOr` I believe could return `undefined` as checked here if the value is a constant, and that value is `undefined` though I'm not sure why that would happen.